Allow void elements to receive marks

SmilinBrian opened the pull request:
**Description**
Some void elements are effectively stand-ins for text, such as with the mentions example, where the mention element renders the character's name. Users might want to format Void elements like this with bold, or set their font and size, so `editor.markableVoid` tells Slate whether or not to apply Marks to the text children of void elements.

- Adds `markableVoid()` as a schema-specific overrideable test.
- Changes `addMark` and `removeMark` so marks can apply to voids. Also changes behavior of collapsed selection so that if a markable Void is selected, the mark will be applied / removed.
- Shows how `markableVoid()` can work in the Mentions example

**Example**
![image](https://user-images.githubusercontent.com/139918/192034205-a6478026-2b8c-43a4-be28-4a643e7ef039.png)
The Mentions example, with R2-D2 in bold text

**Context**
I have achieved a similar effect by overriding `editor.addMark()` and `editor.removeMark()` in our own Slate use case, but we risk missing out on important fixes or changes by having such an override, and this seems generally useful.

**Checks**
- [x] The new code matches the existing patterns and styles.
- [x] The tests pass with `yarn test`.
- [x] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [x] The relevant examples still work. (Run examples with `yarn start`.)
- [x] You've [added a changeset](https://github.com/atlassian/changesets/blob/master/docs/adding-a-changeset.md) if changing functionality. (Add one with `yarn changeset add`.)

dylans reviewed (changes requested):
Could we please decouple the general documentation cleanup into a separate PR? This makes it easier should we need to revert the proposed change in the future.

SmilinBrian commented:
Removed the un-related documentation changes and rebased to combine the new feature into one clear commit.

Sorry this took so long - I was on a cruise ship for a couple weeks so internet was dodgy.

dylans commented:
Thanks! I’ll plan to land this over the weekend unless others have strong suggestions.
